src/YoutubeAtomPlayer.tsx

```diff
@@ -340,6 +353,35 @@ export const YoutubeAtomPlayer = ({
         ],
     );
 
+    /**
+     * Pause the current video when another video is played on the same page
+     */
+    const handlePauseVideo = (
```

Comment:
This function will get re-created on every render of `YoutubeAtomPlayer`.

This is okay if:
- it is not expensive to create
- it is not used by any useEffect, if it is should be made stable
- we can guarantee that there aren't re-renders which avoids any referential equality issues

I think given we are using it in a `useEffect` it will be safer to move the function creation inside the `useEffect` where it will be used.

Reply:
Thanks @arelra for explaining. Could you please elaborate a bit on your third point about referential equality check issues in useEffect?

Reply:
Sure @marjisound, so when we recreate functions inside functional components like this they will get recreated on every render of that component.

That can be okay for the reasons above, however if we use that function in a `useEffect` and it is in its dependency array like:

```js
const myFunction = () => {};
useEffect(() => {
    myFunction();
}, [myFunction]);
```

Every time the component rerenders, `myFunction` is recreated and this `useEffect` will trigger because `useEffect` will see that the reference equality of `myFunction` has changed.

Normally this isn't what you want so if you pass functions to `useEffect` you need to make them stable by either:
- using `useCallback` to memoise the function
- move the function outside of the component
- if the useEffect is called once (i.e. just on component mount) move the function creation inside the useEffect itself

So in this case I think we can move the function creation inside the `useEffect` as we will only trigger it once, because the empty dependency array means it will only run on component mount.

Reply:
Would the values in that function (`videoId`, `player` etc) not then need to be added to the dependancy array?

Reply:
Thanks a lot @arelra  for the detailed explanation. It now makes more sense to me  🙂